### PR TITLE
Milestone 9: Css

### PR DESCRIPTION
**Pull request by Lizdev-05**
I used Capybara to write integration tests for each view in your project. I tested...

- User index page
- user show page
- User post index page
- Post-show page

### CODE REVIEW

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    it "renders the user's post count" do
+      expect(page).to have_content(@user.posts_counter)
+    end
+
+    it "renders the user's bio" do
+      expect(page).to have_content(@user.bio)
+    end
+
+    it "renders the user's last 3 posts" do
```
Comment (tufoinnkuo10) [inline spec/views/user_index_spec.rb]: - [ ] Renders user first 3 post not last. kindly corrrect that

File: spec/views/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      expect(page).to have_content(@user.posts_counter)
+    end
+
+    it 'renders the user show page' do
+      click_link @user.name
+      expect(page).to have_current_path(user_path(@user))
+    end
+  end
+
+  describe 'show page' do
```
Comment (tufoinnkuo10) [inline spec/views/user_index_spec.rb]: - [ ] please the Spec file should only test User index page. the below test for show page should be in its own spec file

File: spec/views/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  it 'Shows the User name' do
+    expect(page).to have_content(@user1)
+  end
+
+  it 'shows number of user posts' do
+    expect(page).to have_content(@user.posts_counter)
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/user_show_spec.rb]: - [ ] Great job. you are missing Tests that asserts you can see a button that lets you view all of a user's posts

File: spec/views/post_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    5.times do |_i|
+      Comment.create(text: 'Nice post!!', author_id: @user.id, post_id: @first_post.id)
+    end
+    visit user_post_path(@first_post.user, @first_post)
+  end
+
+  scenario 'displays the post title' do
+    expect(page).to have_content(@first_post.title)
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/post_show_spec.rb:22]: - [ ] Please add Tests that asserts you can see who wrote the post.

File: spec/views/post_index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      expect(page).to have_content('Next →')
+    end
+
+    it "redirects to a post's show page" do
+      click_link @post1.title
+      expect(page).to have_current_path(user_post_path(@user, @post1))
+    end
+  end
+
+  describe 'show page' do
```
Comment (tufoinnkuo10) [inline spec/views/post_index_spec.rb]: - [ ] The post show page is already tested in its own spec file so its not necesary doing it all over again here. great work

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Lizdev-05 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ Test ok
✅ linter error checks validated
✅ Documented your work professionally


Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/views/user_index_spec.rb (hunk 1 above)
Comment (Lizdev-05) [reply spec/views/user_index_spec.rb]: Thank you for reviewing and correcting my work,. It is not supposed to be the first 3 posts though, but the last ones. the last ones are the recent ones.  I have renamed it to fit the content more though. thank you

File: spec/views/user_index_spec.rb (hunk 2 above)
Comment (Lizdev-05) [reply spec/views/user_index_spec.rb]: Done

File: spec/views/user_show_spec.rb (hunk 3 above)
Comment (Lizdev-05) [reply spec/views/user_show_spec.rb]: It is already there, it tests the route_path. Kindly check again

File: spec/views/post_show_spec.rb (hunk 4 above)
Comment (Lizdev-05) [reply spec/views/post_show_spec.rb:22]: I have a test that confirms the post author, kindly check it well.

File: spec/views/post_index_spec.rb (hunk 5 above)
Comment (Lizdev-05) [reply spec/views/post_index_spec.rb]: Okay.

**Review by descholar-ceo — APPROVED** [review]
Hi @Lizdev-05 🤝  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all of the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**